src/parser/mod.rs

```diff
@@ -7646,6 +7646,39 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_index_exprs(&mut self) -> Result<Vec<IndexExpr>, ParserError> {
+        self.parse_parenthesized(|p| p.parse_comma_separated(Parser::parse_index_expr))
+    }
+
+    pub fn parse_index_expr(&mut self) -> Result<IndexExpr, ParserError> {
+        let expr = self.parse_expr()?;
+        let collation = if self.parse_keyword(Keyword::COLLATE) {
+            Some(self.parse_object_name(false)?)
+        } else {
+            None
+        };
+
+        let (operator_class, order_options) = if self.peek_keyword(Keyword::ASC)
+            || self.peek_keyword(Keyword::DESC)
+            || self.peek_keyword(Keyword::NULLS)
```

Comment:
Instead of using `peek`, we can probably use `self.maybe_parse()` so that we don't need to maintain this list of keywords in sync with the `parse_order_by_options` function e.g.
```rust
let (operator_class, order_options) = if let Some(order_options ) = self.maybe_parse(|p| p.parse_order_by_options())? {
  (None, order_options)
} else {
  let operator_class = self.maybe_parse(|p| p.parse_expr())?;
  let order_options = self.parse_order_by_options()?;
  (operator_class, order_options)
};
```